Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -70,14 +75,122 @@ HTMLAttachmentElement::~HTMLAttachmentElement() = default;
 
 Ref<HTMLAttachmentElement> HTMLAttachmentElement::create(const QualifiedName& tagName, Document& document)
 {
-    return adoptRef(*new HTMLAttachmentElement(tagName, document));
+    Ref<HTMLAttachmentElement> attachment = adoptRef(*new HTMLAttachmentElement(tagName, document));
+    if (DeprecatedGlobalSettings::attachmentWideLayoutEnabled() && DeprecatedGlobalSettings::attachmentElementEnabled()) {
+        ASSERT(attachment->m_implementation == Implementation::Legacy);
+        ASSERT(!attachment->renderer()); // Switch to modern style *must* be done before renderer is created!
+        attachment->m_implementation = Implementation::Modern;
+        if (auto shadow = attachment->userAgentShadowRoot())
+            attachment->ensureModernShadowTree(*shadow);
+        else
+            attachment->ensureUserAgentShadowRoot();
+    }
+    return attachment;
+}
+
+void HTMLAttachmentElement::didAddUserAgentShadowRoot(ShadowRoot& root)
+{
+    if (m_implementation == Implementation::Modern)
+        ensureModernShadowTree(root);
+}
+
+void HTMLAttachmentElement::ensureModernShadowTree(ShadowRoot& root)
+{
+    ASSERT(m_implementation == Implementation::Modern);
+    if (m_elementWithTitle)
+        return;
+
+    constexpr auto styleSheet = R"CSS(
```

Comment:
I saw your comment that said you'd move this to a separate file in a followup, but I don't see why this can't be done now. If you still want to do it separately, please file and bug and leave a `FIXME` here.